core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -217,6 +217,17 @@ public static String temporaryFileName(long nonce, String path) {
   }
 
   /**
+   * @param path the path of the file, possibly temporary
+   * @return the permanent path of the file if it was temporary, or the original path if it was not
+   */
+  public static String getPermanentFileName(String path) {
+    if (isTemporaryFileName(path)) {
+      return path.substring(0, path.length() - 31);
```

Comment:
Yeah I didn't like this either. Do you have any suggestions for calculating the size of the expression? Although our regex is fixed size, most are not so I can see why there is not a size method.